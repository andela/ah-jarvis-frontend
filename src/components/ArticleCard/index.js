/* eslint camelcase: 0 */
import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import likes from '../../assets/icons/thumbs-up.svg';


const Articles = ({ article, index }) => {
  const {
    preview, image, title, slug, likes_count, average_rating,
  } = article;
  const imageComponent = (
    <div className="card-image">
      <Link to={`/article/${slug}`}>
        <img alt="" src={image} className="responsive-img article-img" />
      </Link>
    </div>
  );

  return (
    <div className="card horizontal">
      {index % 2 === 0 && imageComponent }
      <div className="card-stacked">
        <div className="card-content">
          <Link to={`/article/${slug}`} className="black-text">
            <h5 className="truncate">{title}</h5>
          </Link>
          <p>
            <Link to={`/article/${slug}`} className="black-text">
              {`${preview} ... `}
            </Link>
            <Link to={`/article/${slug}`}>
              <u>Read article</u>
            </Link>
          </p>
        </div>
        <div className="card-action card--reactions">
          <div className="stats">
            <span className="icon-stats">
              <img src={likes} alt="likes" />
              {likes_count}
            </span>
            <span className="icon-stats">
              { average_rating }
            </span>
          </div>

        </div>
      </div>
      {index % 2 === 1 && imageComponent }
    </div>
  );
};

Articles.propTypes = {
  index: PropTypes.number.isRequired,
  article: PropTypes.shape().isRequired,
};

export default Articles;
