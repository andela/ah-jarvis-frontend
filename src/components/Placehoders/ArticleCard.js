import React from 'react';

import ContentLoader from 'react-content-loader';

const ArticleCardPlaceHolder = props => (
  <ContentLoader
    height={100}
    width={400}
    speed={1.5}
    primaryColor="#dfdfdf"
    secondaryColor="#ffffff"
    {...props}
  >
    <rect x="165" y="13" rx="5" ry="5" width="220" height="4.6" />
    <rect x="166" y="29.05" rx="5" ry="5" width="220" height="4.5" />
    <rect x="165" y="44" rx="5" ry="5" width="220" height="4.6" />
    <rect x="165" y="64.05" rx="5" ry="5" width="220" height="4.58" />
    <rect x="-7" y="0.05" rx="0" ry="0" width="158.4" height="117.97360000000002" />
    <rect x="319" y="17.05" rx="0" ry="0" width="4" height="0" />
    <rect x="293" y="17.05" rx="0" ry="0" width="0" height="0" />
    <rect x="269" y="19.05" rx="0" ry="0" width="0" height="0" />
    <rect x="220" y="17.05" rx="0" ry="0" width="0" height="0" />
    <rect x="219" y="125.05" rx="0" ry="0" width="0" height="0" />
    <rect x="233" y="121.05" rx="0" ry="0" width="0" height="0" />
    <rect x="164" y="76.05" rx="0" ry="0" width="211" height="1" />
    <circle cx="175.6" cy="88.95" r="3.5" />
    <circle cx="198.84" cy="88.95" r="3.5" />
  </ContentLoader>
);

export default ArticleCardPlaceHolder;
