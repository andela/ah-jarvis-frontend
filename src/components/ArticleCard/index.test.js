import { shallow } from 'enzyme';
import React from 'react';
import ArticleCard from '.';

describe('<ArticleCard />', () => {
  it('renders  <ArticleCard /> components', () => {
    const snap = shallow(<ArticleCard article={{
      preview: 'cool', image: '', title: 'cool', slug: 'coll', likes_count: '', average_rating: '',
    }}
    />);
    expect(snap).toMatchSnapshot();
  });
});
