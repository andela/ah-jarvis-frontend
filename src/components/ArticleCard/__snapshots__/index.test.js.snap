// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArticleCard /> renders  <ArticleCard /> components 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Articles
    article={
        Object {
            "average_rating": "",
            "image": "",
            "likes_count": "",
            "preview": "cool",
            "slug": "coll",
            "title": "cool",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        false,
        <div
          className="card-stacked"
>
          <div
                    className="card-content"
          >
                    <Link
                              className="black-text"
                              replace={false}
                              to="/article/coll"
                    >
                              <h5
                                        className="truncate"
                              >
                                        cool
                              </h5>
                    </Link>
                    <p>
                              <Link
                                        className="black-text"
                                        replace={false}
                                        to="/article/coll"
                              >
                                        cool ... 
                              </Link>
                              <Link
                                        replace={false}
                                        to="/article/coll"
                              >
                                        <u>
                                                  Read article
                                        </u>
                              </Link>
                    </p>
          </div>
          <div
                    className="card-action card--reactions"
          >
                    <div
                              className="stats"
                    >
                              <div
                                        className="icon---default"
                              >
                                        <img
                                                  alt="thumbs-up"
                                                  className="p-r--5"
                                                  src="thumbs-up.svg"
                                        />
                                        <span
                                                  className="text--small"
                                        />
                              </div>
                              <span
                                        className="icon-stats"
                              />
                    </div>
          </div>
</div>,
        false,
      ],
      "className": "card horizontal",
    },
    "ref": null,
    "rendered": Array [
      false,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="card-content"
>
              <Link
                            className="black-text"
                            replace={false}
                            to="/article/coll"
              >
                            <h5
                                          className="truncate"
                            >
                                          cool
                            </h5>
              </Link>
              <p>
                            <Link
                                          className="black-text"
                                          replace={false}
                                          to="/article/coll"
                            >
                                          cool ... 
                            </Link>
                            <Link
                                          replace={false}
                                          to="/article/coll"
                            >
                                          <u>
                                                        Read article
                                          </u>
                            </Link>
              </p>
</div>,
            <div
              className="card-action card--reactions"
>
              <div
                            className="stats"
              >
                            <div
                                          className="icon---default"
                            >
                                          <img
                                                        alt="thumbs-up"
                                                        className="p-r--5"
                                                        src="thumbs-up.svg"
                                          />
                                          <span
                                                        className="text--small"
                                          />
                            </div>
                            <span
                                          className="icon-stats"
                            />
              </div>
</div>,
          ],
          "className": "card-stacked",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Link
                  className="black-text"
                  replace={false}
                  to="/article/coll"
>
                  <h5
                                    className="truncate"
                  >
                                    cool
                  </h5>
</Link>,
                <p>
                  <Link
                                    className="black-text"
                                    replace={false}
                                    to="/article/coll"
                  >
                                    cool ... 
                  </Link>
                  <Link
                                    replace={false}
                                    to="/article/coll"
                  >
                                    <u>
                                                      Read article
                                    </u>
                  </Link>
</p>,
              ],
              "className": "card-content",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <h5
                    className="truncate"
>
                    cool
</h5>,
                  "className": "black-text",
                  "replace": false,
                  "to": "/article/coll",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "cool",
                    "className": "truncate",
                  },
                  "ref": null,
                  "rendered": "cool",
                  "type": "h5",
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <Link
                      className="black-text"
                      replace={false}
                      to="/article/coll"
>
                      cool ... 
</Link>,
                    <Link
                      replace={false}
                      to="/article/coll"
>
                      <u>
                                            Read article
                      </u>
</Link>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "cool ... ",
                      "className": "black-text",
                      "replace": false,
                      "to": "/article/coll",
                    },
                    "ref": null,
                    "rendered": "cool ... ",
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": <u>
                        Read article
</u>,
                      "replace": false,
                      "to": "/article/coll",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Read article",
                      },
                      "ref": null,
                      "rendered": "Read article",
                      "type": "u",
                    },
                    "type": [Function],
                  },
                ],
                "type": "p",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="stats"
>
                <div
                                className="icon---default"
                >
                                <img
                                                alt="thumbs-up"
                                                className="p-r--5"
                                                src="thumbs-up.svg"
                                />
                                <span
                                                className="text--small"
                                />
                </div>
                <span
                                className="icon-stats"
                />
</div>,
              "className": "card-action card--reactions",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="icon---default"
>
                    <img
                                        alt="thumbs-up"
                                        className="p-r--5"
                                        src="thumbs-up.svg"
                    />
                    <span
                                        className="text--small"
                    />
</div>,
                  <span
                    className="icon-stats"
/>,
                ],
                "className": "stats",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <img
                        alt="thumbs-up"
                        className="p-r--5"
                        src="thumbs-up.svg"
/>,
                      <span
                        className="text--small"
/>,
                    ],
                    "className": "icon---default",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "alt": "thumbs-up",
                        "className": "p-r--5",
                        "src": "thumbs-up.svg",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "img",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "",
                        "className": "text--small",
                      },
                      "ref": null,
                      "rendered": "",
                      "type": "span",
                    },
                  ],
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "",
                    "className": "icon-stats",
                  },
                  "ref": null,
                  "rendered": "",
                  "type": "span",
                },
              ],
              "type": "div",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      false,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          false,
          <div
            className="card-stacked"
>
            <div
                        className="card-content"
            >
                        <Link
                                    className="black-text"
                                    replace={false}
                                    to="/article/coll"
                        >
                                    <h5
                                                className="truncate"
                                    >
                                                cool
                                    </h5>
                        </Link>
                        <p>
                                    <Link
                                                className="black-text"
                                                replace={false}
                                                to="/article/coll"
                                    >
                                                cool ... 
                                    </Link>
                                    <Link
                                                replace={false}
                                                to="/article/coll"
                                    >
                                                <u>
                                                            Read article
                                                </u>
                                    </Link>
                        </p>
            </div>
            <div
                        className="card-action card--reactions"
            >
                        <div
                                    className="stats"
                        >
                                    <div
                                                className="icon---default"
                                    >
                                                <img
                                                            alt="thumbs-up"
                                                            className="p-r--5"
                                                            src="thumbs-up.svg"
                                                />
                                                <span
                                                            className="text--small"
                                                />
                                    </div>
                                    <span
                                                className="icon-stats"
                                    />
                        </div>
            </div>
</div>,
          false,
        ],
        "className": "card horizontal",
      },
      "ref": null,
      "rendered": Array [
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="card-content"
>
                <Link
                                className="black-text"
                                replace={false}
                                to="/article/coll"
                >
                                <h5
                                                className="truncate"
                                >
                                                cool
                                </h5>
                </Link>
                <p>
                                <Link
                                                className="black-text"
                                                replace={false}
                                                to="/article/coll"
                                >
                                                cool ... 
                                </Link>
                                <Link
                                                replace={false}
                                                to="/article/coll"
                                >
                                                <u>
                                                                Read article
                                                </u>
                                </Link>
                </p>
</div>,
              <div
                className="card-action card--reactions"
>
                <div
                                className="stats"
                >
                                <div
                                                className="icon---default"
                                >
                                                <img
                                                                alt="thumbs-up"
                                                                className="p-r--5"
                                                                src="thumbs-up.svg"
                                                />
                                                <span
                                                                className="text--small"
                                                />
                                </div>
                                <span
                                                className="icon-stats"
                                />
                </div>
</div>,
            ],
            "className": "card-stacked",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <Link
                    className="black-text"
                    replace={false}
                    to="/article/coll"
>
                    <h5
                                        className="truncate"
                    >
                                        cool
                    </h5>
</Link>,
                  <p>
                    <Link
                                        className="black-text"
                                        replace={false}
                                        to="/article/coll"
                    >
                                        cool ... 
                    </Link>
                    <Link
                                        replace={false}
                                        to="/article/coll"
                    >
                                        <u>
                                                            Read article
                                        </u>
                    </Link>
</p>,
                ],
                "className": "card-content",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": <h5
                      className="truncate"
>
                      cool
</h5>,
                    "className": "black-text",
                    "replace": false,
                    "to": "/article/coll",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "cool",
                      "className": "truncate",
                    },
                    "ref": null,
                    "rendered": "cool",
                    "type": "h5",
                  },
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <Link
                        className="black-text"
                        replace={false}
                        to="/article/coll"
>
                        cool ... 
</Link>,
                      <Link
                        replace={false}
                        to="/article/coll"
>
                        <u>
                                                Read article
                        </u>
</Link>,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "children": "cool ... ",
                        "className": "black-text",
                        "replace": false,
                        "to": "/article/coll",
                      },
                      "ref": null,
                      "rendered": "cool ... ",
                      "type": [Function],
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "children": <u>
                          Read article
</u>,
                        "replace": false,
                        "to": "/article/coll",
                      },
                      "ref": null,
                      "rendered": Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "Read article",
                        },
                        "ref": null,
                        "rendered": "Read article",
                        "type": "u",
                      },
                      "type": [Function],
                    },
                  ],
                  "type": "p",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <div
                  className="stats"
>
                  <div
                                    className="icon---default"
                  >
                                    <img
                                                      alt="thumbs-up"
                                                      className="p-r--5"
                                                      src="thumbs-up.svg"
                                    />
                                    <span
                                                      className="text--small"
                                    />
                  </div>
                  <span
                                    className="icon-stats"
                  />
</div>,
                "className": "card-action card--reactions",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <div
                      className="icon---default"
>
                      <img
                                            alt="thumbs-up"
                                            className="p-r--5"
                                            src="thumbs-up.svg"
                      />
                      <span
                                            className="text--small"
                      />
</div>,
                    <span
                      className="icon-stats"
/>,
                  ],
                  "className": "stats",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        <img
                          alt="thumbs-up"
                          className="p-r--5"
                          src="thumbs-up.svg"
/>,
                        <span
                          className="text--small"
/>,
                      ],
                      "className": "icon---default",
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "alt": "thumbs-up",
                          "className": "p-r--5",
                          "src": "thumbs-up.svg",
                        },
                        "ref": null,
                        "rendered": null,
                        "type": "img",
                      },
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "",
                          "className": "text--small",
                        },
                        "ref": null,
                        "rendered": "",
                        "type": "span",
                      },
                    ],
                    "type": "div",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "",
                      "className": "icon-stats",
                    },
                    "ref": null,
                    "rendered": "",
                    "type": "span",
                  },
                ],
                "type": "div",
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
        false,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
